### DISCUSSION TITLE
Support for different touch controller, please.

### DISCUSSION THREAD

**Discussion by MikeyMoMo** [Bodmer/TFT_eSPI#1872] [Compatible support libraries]
I have a beautiful display.  3.5 inch capacitive touch using a FT6336U touch controller.  The code to support this is already written and is in the LovyanGFX library.  Could you possibly consider adding that touch code in to your library?

The display works just fine with your library.  I just need touch to work.  I have 5 kinds of displays and use TFT_eSPI on all of them.  This 5th one has this different touch controller.  I would like to keep it standard.  The 4 inch one I have uses your already included touch controller just fine.

My problem is that the program is consuming a lot of memory (2,167,633 bytes (68%) of program storage space).  I have already switched to Huge App in the Partition Scheme.  I would not like to have to pull in the LovyanGFX to only get touch from it.  It is based on your code, it seems, and has the same routines names so it would take a lot of editing to send the display calls to you and the touch calls to it. 
And, when memory usage gets high, programs get unstable.  I guess I could switch over to that library, totally, but I am using your sprite code and who knows how that would work and how long to make all the changes needed to move to a different driver library.  I am trying to keep the code as similar as possible between the various displays it runs on. (Turns out the Sprite code requirement is different)

Of course, it would take a little setup file to select the correct touch driver, similar to the display selection files.  So a little structure change would be needed unless there is some way to autodetect.  I am not skilled in that part of this stuff.  Just guessing.

Here is all I have on it:

> Touch (FT6336U)
> 
> TOUCH_SDA=18
> TOUCH_SCL=19
> I2C_TOUCH_ADDRESS=0x38

Either way, thanks for all the work you have put into this library.  I use it 24 hours a day.

Mike

**Comment by Bodmer** [comment]
There are supporting Arduino libraries for that touch controller, for example here:

https://github.com/aselectroworks/Arduino-FT6336U

As I do not have a screen with that touch controller it would be rather hit and miss getting it integrated into TFT_ESPI and at the moemnt I have paid for projects to do!

**Comment by MikeyMoMo** [comment]
Ok.  Thanks.  I did not know about the stand-alone libraries for it.  Will
go that way.  Stupid me. Never looked.  There seems to be enough interest
in this board to finance one for you, if the time arrises to include that
code.

You can close this or I will try to remember.  Making money is good.  Go to
it!
